File: 백준/Bronze/6679. 싱기한 네자리 숫자/싱기한 네자리 숫자.js
for (let i = 1000; i <= 9999; i++) {
  let sixteen = i;
  let sixteenSum = 0;
  while (sixteen != 0) {
    sixteenSum += sixteen % 16;
    sixteen = Math.floor(sixteen / 16);
  }

  let twelve = i;
  let twelveSum = 0;
  while (twelve != 0) {
    twelveSum += twelve % 12;
    twelve = Math.floor(twelve / 12);
  }

  let ten = i;
  let tenSum = 0;
  while (ten != 0) {
    tenSum += ten % 10;
    ten = Math.floor(ten / 10);
  }

  if (sixteenSum === twelveSum && twelveSum === tenSum) {
    console.log(i);
  }
}

File: 백준/Bronze/6679. 싱기한 네자리 숫자/싱기한 네자리 숫자.py
for i in range(1000, 10000):
    sixteen = twelve = ten = i
    sixteenSum = twelveSum = tenSum = 0
    while sixteen != 0:
        sixteenSum += sixteen % 16
        sixteen //= 16
    while twelve != 0:
        twelveSum += twelve % 12
        twelve //= 12
    while ten != 0:
        tenSum += ten % 10
        ten //= 10
    
    if sixteenSum == twelveSum == tenSum:
        print(i)